loadtest/control/simulcontroller/drafts.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package simulcontroller
+
+import (
+	"errors"
+	"fmt"
+	"math/rand"
+
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/control"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/store/memstore"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/user"
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+func (c *SimulController) getDrafts(u user.User) control.UserActionResponse {
+	if ok, resp := control.DraftsEnabled(u); resp.Err != nil {
+		return resp
+	} else if !ok {
+		return control.UserActionResponse{Info: "drafts not enabled"}
+	}
+
+	userId := u.Store().Id()
+
+	team, err := u.Store().CurrentTeam()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	} else if team == nil {
+		return control.UserActionResponse{Err: control.NewUserError(errors.New("current team should be set"))}
+	}
+
+	err = u.GetDrafts(team.Id)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	return control.UserActionResponse{Info: fmt.Sprintf("viewed drafts for user id %v in team id %v", userId, team.Id)}
+}
+
+func (c *SimulController) upsertDraft(u user.User) control.UserActionResponse {
+	if ok, resp := control.DraftsEnabled(u); resp.Err != nil {
+		return resp
+	} else if !ok {
+		return control.UserActionResponse{Info: "drafts not enabled"}
+	}
+
+	channel, err := u.Store().CurrentChannel()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	var rootId = ""
+
+	// 33% of the time draft will be a thread reply
+	// NOTE: This needs to be verified.
```

Comment:
We can check this with a DB query right?

Reply:
Yep, good idea.

Reply:
done